### PR TITLE
Update pytest and pytest-* to current versions

### PR DESCRIPTION
**Pull request by bgermann**
Pytest 6 was released recently with a lot of improvements and bug fixes.
Use the latest pytest version and the latest pytest plugin versions which
might have changes to be compatible with the 6.x versions.

**Description**

**Checklist**

- [x] PR has been tested

### CODE REVIEW

**Comment by Emantor** [comment]
From the coverage diff it looks like the `junitxml` usage we currently use is no longer valid. This should be fixed before moving to pytest 6.x, since this outputs a jenkins compatible XML.

**Comment by bgermann** [comment]
I think this should be fixed.

File: labgrid/pytestplugin/fixtures.py [hunk 1]
```diff
@@ -60,49 +60,47 @@ def env(request):
         pytest.skip("missing environment config (use --lg-env)")
 
     if request.config.pluginmanager.hasplugin('junitxml'):
```
Comment (Emantor) [inline labgrid/pytestplugin/fixtures.py:62]: According to the source code at [pytest](https://docs.pytest.org/en/stable/_modules/_pytest/junitxml.html#record_testsuite_property), this should no longer be necessary, the fixture will return a no-op function in this case.

**Review by Emantor — CHANGES_REQUESTED** [review]
This looks good, thank you!
With the removal mentioned in the comments and a bump of the minimal pytest version in `setup.py` this should be good to go.

**Review by Emantor — APPROVED** [review]

**Pull request merged**